csrc/loader.cpp

```diff
@@ -25,6 +25,11 @@
 // 0x1010107f == v1.1.1g release
 #define LIBCRYPTO_MAJOR_MINOR_VERSION_MASK 0xFFF00000
 
+#define LIBCRYPTO_EXACT_VERSION_MATCH       0
+#define LIBCRYPTO_FUZZY_VERSION_MATCH       1
+
+static char accp_loader_exception_msg[256] = {0};
```

Comment:
This is technically not thread safe anymore. Could you move this back (non statically) inside `accpValidateLibcryptoVersion` please?

Reply:
does throwing an exception result in copying exception data over to the JVM heap or some other location? if not, and we stack-allocate the message buffer within the scope of `accpValidateLibcryptoVersion` method, de-referencing the buffer outside of `accpValidateLibcryptoVersion`'s scope would violate the buffer's lifetime.

Reply:
Oh dear. You're right.

Hypothetically this could be a bigger issue. We should probably rewrite `java_ex` to always (and immediately) copy the string into local memory (see `env.h`, it should be easy). Can we pull that into this change as there really isn't a safe path forward without that?

Reply:
Fixed. `java_ex` now keeps a local copy of the message passed to it.